'sprintf' Depricated on MacOs

| | |
|:---|:---|
|**_F´ Version_**| Devel |
|**_Affected Component_**| Health Component Unit Tests |
---
## Problem Description

When checking all unit tests on the top level directory on a MacOs machine, a build error occurs. The compiler complains that `sprintf` is deprecated and should not be used. This error prevents the build from continuing.

## How to Reproduce

1. `fprime-util generate --ut`
2. `fprime-util check --all`

## Expected Behavior

Here is the expected error:

```
/Users/lsoares/Documents/fprime/Svc/Health/test/ut/Tester.cpp:263:11: error: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Werror,-Wdeprecated-declarations]
          sprintf(name,"task%d",port);
          ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.0.sdk/usr/include/stdio.h:188:1: note: 'sprintf' has been explicitly marked deprecated here
__deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.0.sdk/usr/include/sys/cdefs.h:215:48: note: expanded from macro '__deprecated_msg'
        #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                      ^
/Users/lsoares/Documents/fprime/Svc/Health/test/ut/Tester.cpp:297:11: error: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Werror,-Wdeprecated-declarations]
          sprintf(name,"task%d",port);
          ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.0.sdk/usr/include/stdio.h:188:1: note: 'sprintf' has been explicitly marked deprecated here
__deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.0.sdk/usr/include/sys/cdefs.h:215:48: note: expanded from macro '__deprecated_msg'
        #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                      ^
/Users/lsoares/Documents/fprime/Svc/Health/test/ut/Tester.cpp:319:11: error: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Werror,-Wdeprecated-declarations]
          sprintf(name,"task%d",port);
          ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.0.sdk/usr/include/stdio.h:188:1: note: 'sprintf' has been explicitly marked deprecated here
__deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.0.sdk/usr/include/sys/cdefs.h:215:48: note: expanded from macro '__deprecated_msg'
        #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                      ^
/Users/lsoares/Documents/fprime/Svc/Health/test/ut/Tester.cpp:349:11: error: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Werror,-Wdeprecated-declarations]
          sprintf(name,"task%d",entry);
          ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.0.sdk/usr/include/stdio.h:188:1: note: 'sprintf' has been explicitly marked deprecated here
__deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.0.sdk/usr/include/sys/cdefs.h:215:48: note: expanded from macro '__deprecated_msg'
        #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                      ^
/Users/lsoares/Documents/fprime/Svc/Health/test/ut/Tester.cpp:409:11: error: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Werror,-Wdeprecated-declarations]
          sprintf(name,"task%d",entry);
          ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.0.sdk/usr/include/stdio.h:188:1: note: 'sprintf' has been explicitly marked deprecated here
__deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.0.sdk/usr/include/sys/cdefs.h:215:48: note: expanded from macro '__deprecated_msg'
        #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                      ^
/Users/lsoares/Documents/fprime/Svc/Health/test/ut/Tester.cpp:438:11: error: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Werror,-Wdeprecated-declarations]
          sprintf(name,"task%d",entry);
          ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.0.sdk/usr/include/stdio.h:188:1: note: 'sprintf' has been explicitly marked deprecated here
__deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.0.sdk/usr/include/sys/cdefs.h:215:48: note: expanded from macro '__deprecated_msg'
        #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                      ^
/Users/lsoares/Documents/fprime/Svc/Health/test/ut/Tester.cpp:498:11: error: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Werror,-Wdeprecated-declarations]
          sprintf(name,"task%d",entry);
          ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.0.sdk/usr/include/stdio.h:188:1: note: 'sprintf' has been explicitly marked deprecated here
__deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.0.sdk/usr/include/sys/cdefs.h:215:48: note: expanded from macro '__deprecated_msg'
        #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                      ^
/Users/lsoares/Documents/fprime/Svc/Health/test/ut/Tester.cpp:659:11: error: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Werror,-Wdeprecated-declarations]
          sprintf(name,"task%d",port);
          ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.0.sdk/usr/include/stdio.h:188:1: note: 'sprintf' has been explicitly marked deprecated here
__deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.0.sdk/usr/include/sys/cdefs.h:215:48: note: expanded from macro '__deprecated_msg'
        #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                      ^
/Users/lsoares/Documents/fprime/Svc/Health/test/ut/Tester.cpp:751:7: error: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Werror,-Wdeprecated-declarations]
      sprintf(name,"task%d",Svc::HealthComponentBase::NUM_PINGSEND_OUTPUT_PORTS-1);
      ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.0.sdk/usr/include/stdio.h:188:1: note: 'sprintf' has been explicitly marked deprecated here
__deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.0.sdk/usr/include/sys/cdefs.h:215:48: note: expanded from macro '__deprecated_msg'
        #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                      ^
9 errors generated.
make[3]: *** [F-Prime/Svc/Health/CMakeFiles/Svc_Health_ut_exe.dir/test/ut/Tester.cpp.o] Error 1
```

--- a/Svc/ActiveTextLogger/test/ut/Tester.cpp
+++ b/Svc/ActiveTextLogger/test/ut/Tester.cpp
@@ -103,7 +103,7 @@ namespace Svc {
           if (stream1) {
               std::cout << "readLine: " << buf << std::endl;
               char textStr[512];
-              sprintf(textStr,
+              snprintf(textStr, sizeof(textStr),
                       "EVENT: (%d) (%d:%d,%d) %s: %s",
                        id,timeTag.getTimeBase(),timeTag.getSeconds(),timeTag.getUSeconds(),severityString,text.toChar());
               ASSERT_EQ(0,strcmp(textStr,buf));
@@ -144,7 +144,7 @@ namespace Svc {
               // Verify new printed line
               else {
                   char textStr[512];
-                  sprintf(textStr,
+                  snprintf(textStr, sizeof(textStr),
                           "EVENT: (%d) (%d:%d,%d) %s: %s",
                            id,timeTag.getTimeBase(),timeTag.getSeconds(),timeTag.getUSeconds(),severityString,text.toChar());
                   ASSERT_EQ(0,strcmp(textStr,buf));
@@ -206,7 +206,7 @@ namespace Svc {
           if (stream1) {
               std::cout << "readLine: " << buf << std::endl;
               char textStr[512];
-              sprintf(textStr,
+              snprintf(textStr, sizeof(textStr),
                       "EVENT: (%d) (%d:%d,%d) %s: %s",
                        id,timeTag.getTimeBase(),timeTag.getSeconds(),timeTag.getUSeconds(),severityString,text.toChar());
               ASSERT_EQ(0,strcmp(textStr,buf));
@@ -316,7 +316,7 @@ namespace Svc {
 
           stat = this->component.set_log_file(baseName,50);
 
-          sprintf(baseNameWithSuffix,"%s%d",baseName,i);
+          snprintf(baseNameWithSuffix, sizeof(baseNameWithSuffix), "%s%d",baseName,i);
           ASSERT_TRUE(stat);
           ASSERT_TRUE(this->component.m_log_file.m_openFile);
           ASSERT_EQ(0,strcmp(baseNameWithSuffix,this->component.m_log_file.m_fileName.toChar()));
@@ -327,7 +327,7 @@ namespace Svc {
       // Create 11th which will fail and re-use the original:
       stat = this->component.set_log_file(baseName,50);
 
-      sprintf(baseNameWithSuffix,"%s%d",baseName,i);
+      snprintf(baseNameWithSuffix, sizeof(baseNameWithSuffix), "%s%d",baseName,i);
       ASSERT_TRUE(stat);
       ASSERT_TRUE(this->component.m_log_file.m_openFile);
       printf("<< %s %s\n",baseName,this->component.m_log_file.m_fileName.toChar());
@@ -341,7 +341,7 @@ namespace Svc {
       remove(longFileNameDup);
       remove(baseName);
       for (i = 0; i < 10; ++i) {
-          sprintf(baseNameWithSuffix,"%s%d",baseName,i);
+          snprintf(baseNameWithSuffix, sizeof(baseNameWithSuffix), "%s%d",baseName,i);
           remove(baseNameWithSuffix);
       }
 

--- a/Svc/Health/test/ut/Tester.cpp
+++ b/Svc/Health/test/ut/Tester.cpp
@@ -260,7 +260,7 @@ namespace Svc {
 
       for (NATIVE_UINT_TYPE port = 0; port < Svc::HealthComponentBase::NUM_PINGSEND_OUTPUT_PORTS; port++) {
           char name[80];
-          sprintf(name,"task%d",port);
+          snprintf(name, sizeof(name), "task%d",port);
           ASSERT_EVENTS_HLTH_PING_WARN(port,name);
       }
 
@@ -294,7 +294,7 @@ namespace Svc {
 
       for (NATIVE_UINT_TYPE port = 0; port < Svc::HealthComponentBase::NUM_PINGSEND_OUTPUT_PORTS; port++) {
           char name[80];
-          sprintf(name,"task%d",port);
+          snprintf(name, sizeof(name), "task%d",port);
           ASSERT_EVENTS_HLTH_PING_WARN(port,name);
       }
 
@@ -316,7 +316,7 @@ namespace Svc {
 
       for (NATIVE_UINT_TYPE port = 0; port < Svc::HealthComponentBase::NUM_PINGSEND_OUTPUT_PORTS; port++) {
           char name[80];
-          sprintf(name,"task%d",port);
+          snprintf(name, sizeof(name), "task%d",port);
           ASSERT_EVENTS_HLTH_PING_LATE(port,name);
       }
 
@@ -346,7 +346,7 @@ namespace Svc {
       ASSERT_EVENTS_HLTH_PING_WARN_SIZE(Svc::HealthComponentBase::NUM_PINGSEND_OUTPUT_PORTS);
       for (NATIVE_INT_TYPE entry = 0; entry < Svc::HealthComponentBase::NUM_PINGSEND_OUTPUT_PORTS; entry++) {
           char name[80];
-          sprintf(name,"task%d",entry);
+          snprintf(name, sizeof(name), "task%d",entry);
           ASSERT_EVENTS_HLTH_PING_WARN(entry, name);
       }
 
@@ -406,7 +406,7 @@ namespace Svc {
       ASSERT_EVENTS_HLTH_PING_LATE_SIZE(Svc::HealthComponentBase::NUM_PINGSEND_OUTPUT_PORTS);
       for (NATIVE_INT_TYPE entry = 0; entry < Svc::HealthComponentBase::NUM_PINGSEND_OUTPUT_PORTS; entry++) {
           char name[80];
-          sprintf(name,"task%d",entry);
+          snprintf(name,sizeof(name), "task%d",entry);
           ASSERT_EVENTS_HLTH_PING_LATE(entry,name);
       }
 
@@ -435,7 +435,7 @@ namespace Svc {
           }
           // disable entry
           char name[80];
-          sprintf(name,"task%d",entry);
+          snprintf(name, sizeof(name), "task%d",entry);
           Fw::CmdStringArg task(name);
           this->sendCmd_HLTH_PING_ENABLE(0,10,name,Fw::Enabled::DISABLED);
           this->dispatchAll();
@@ -495,7 +495,7 @@ namespace Svc {
           ASSERT_EQ(this->pingEntries[entry].warnCycles,Svc::HealthComponentBase::NUM_PINGSEND_OUTPUT_PORTS+entry);
           ASSERT_EQ(this->pingEntries[entry].fatalCycles,Svc::HealthComponentBase::NUM_PINGSEND_OUTPUT_PORTS*2+entry+1);
           char name[80];
-          sprintf(name,"task%d",entry);
+          snprintf(name, sizeof(name), "task%d",entry);
           Fw::CmdStringArg task(name);
           this->sendCmd_HLTH_CHNG_PING(0,10,task,
                   Svc::HealthComponentBase::NUM_PINGSEND_OUTPUT_PORTS*4+entry,
@@ -656,7 +656,7 @@ namespace Svc {
       ASSERT_EVENTS_HLTH_PING_WRONG_KEY_SIZE(Svc::HealthComponentBase::NUM_PINGSEND_OUTPUT_PORTS);
       for (NATIVE_INT_TYPE port = 0; port < Svc::HealthComponentBase::NUM_PINGSEND_OUTPUT_PORTS; port++) {
           char name[80];
-          sprintf(name,"task%d",port);
+          snprintf(name, sizeof(name), "task%d",port);
           ASSERT_EVENTS_HLTH_PING_WRONG_KEY(port,name,FLAG_KEY_VALUE);
       }
 
@@ -748,7 +748,7 @@ namespace Svc {
       ASSERT_EVENTS_SIZE(1);
       ASSERT_EVENTS_HLTH_PING_WARN_SIZE(1);
       char name[80];
-      sprintf(name,"task%d",Svc::HealthComponentBase::NUM_PINGSEND_OUTPUT_PORTS-1);
+      snprintf(name, sizeof(name), "task%d",Svc::HealthComponentBase::NUM_PINGSEND_OUTPUT_PORTS-1);
       ASSERT_EVENTS_HLTH_PING_WARN(0,name);
       ASSERT_TLM_SIZE(1);
       ASSERT_TLM_PingLateWarnings_SIZE(1);

